src/db.c

```diff
@@ -582,10 +611,12 @@ long long emptyDbStructure(serverDb **dbarray, int dbnum, int async, void(callba
         } else {
             kvstoreEmpty(dbarray[j]->keys, callback);
             kvstoreEmpty(dbarray[j]->expires, callback);
+            kvstoreEmpty(dbarray[j]->keys_with_volatile_items, callback);
         }
         /* Because all keys of database are removed, reset average ttl. */
         dbarray[j]->avg_ttl = 0;
         dbarray[j]->expires_cursor = 0;
+        dbarray[j]->keys_with_volatile_items_cursor = 0;
```

Comment:
nit: this might be better to wrap around a function like "resetExpiryCycle" or something. 
We can place all the avg_ttl, expires_cursor and keys_with_volatile_items_cursor resets and the code would be clearer IMO.

Reply:
added a helper function.